### PR TITLE
Preparation for 2024 election

### PR DESCRIPTION
**Pull request by axelg12**
Making changes for new questions and use of SVT election compass 

**Changes so far**

- draft of new questions
- Usage of SVT election compass to calculate new score

A small update of how the changes will look

image

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -20,12 +20,14 @@
     "classnames": "^2.2.6",
     "core-js": "^2.6.11",
     "debug": "^3.2.6",
+    "election-compass-match": "^2.0.0",
```
Comment (koddsson) [inline package.json:23]: This is a cool library!

File: src/svt-process-replies.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+      if (answer === '6') {
+        return null;
+      }
+      // SVT uses 4-level Likert scale by default while we use 5 level
+      // To make it work with the 5-level scale we need to set the type as range
+      if (Math.random() < 0.5) {
+        // Used to test out the scale
+        return { selectedIndex: answer - 1, isImportant: true, type: 'RANGE' };
+      }
+      return { selectedIndex: answer - 1, isImportant: false, type: 'RANGE' };
```
Comment (koddsson) [inline src/svt-process-replies.js:34]: What do you think about



```suggestion
      return { 
        selectedIndex: answer - 1,
        // Used to test out the scale
        isImportant: Math.random() < 0.5,
        type: 'RANGE' 
      };
```

File: data/poll/answers.json [hunk 3]
```diff
@@ -1,10 +1,10 @@
...
-    "1": "Mjög ósammála",
-    "2": "Frekar ósammála",
-    "3": "Hlutlaus",
-    "4": "Frekar sammála",
-    "5": "Mjög sammála"
+    "0": "Mjög ósammála",
+    "1": "Frekar ósammála",
+    "2": "Hlutlaus",
+    "3": "Frekar sammála",
+    "4": "Mjög sammála"
```
Comment (koddsson) [inline data/poll/answers.json:8]: This could potentially be a array at this point, right?

File: package.json [hunk 4]
```diff
@@ -20,12 +20,14 @@
...
     "express": "^4.17.1",
     "history": "^4.10.1",
     "ioredis": "^4.27.9",
     "isomorphic-style-loader": "^3.0.0",
     "kennitala": "^1.2.4",
     "lodash": "^4.17.20",
+    "lzjs": "^1.3.1",
```
Comment (koddsson) [inline package.json:30]: I don't see this being used in the rest of the diff.

File: src/process-replies.js [hunk 5]
```diff
@@ -6,7 +6,7 @@ const valueMap = {
   '3': 0,
   '4': 0.8,
   '5': 1,
-  '6': null
+  '6': null,
```
Comment (koddsson) [inline src/process-replies.js:9]: I guess this is prettier running in your editor making these changes. Maybe we should add prettier to the project and run it in CI so we don't have unrelated changes like this in the PRs?

File: src/routes/prof-nidurstodur/KosningaProf.js [hunk 6]
```diff
@@ -19,9 +19,15 @@ const initialAnswers = questions =>
...
   3: 'Hlutlaus',
-  5: 'Mjög sammála',
+  4: 'Mjög sammála',
+};
+
+const IMPORTANT_SYMBOL = '!';
+
+const isImportant = answer => {
+  return answer.includes(IMPORTANT_SYMBOL);
 };
```
Comment (koddsson) [inline src/routes/prof-nidurstodur/KosningaProf.js:31]: Maybe this is just:

```suggestion
const isImportant = answer => {
  return answer.includes('!');
};
```

?

**Review by koddsson — APPROVED** [review]
I hope it's ok that I do a drive-by review 👯

**Comment by axelg12** [comment]
> I hope it's ok that I do a drive-by review 👯

Of course!  They are welcomed

File: src/svt-process-replies.js (hunk 2 above)
Comment (axelg12) [reply src/svt-process-replies.js:34]: Fine by me, but we should remove this before we merge it to master, given we have the new questions ready

File: package.json (hunk 4 above)
Comment (axelg12) [reply package.json:30]: great catch

File: src/process-replies.js (hunk 5 above)
Comment (axelg12) [reply src/process-replies.js:9]: Great point. I will do that in a separate PR

Comment (axelg12) [reply src/process-replies.js:9]: https://github.com/kristjanmik/kjosturett-web/pull/69

**Pull request merged**